Userland/Services/WebDriver/Session.cpp

```diff
@@ -106,4 +114,55 @@ Web::WebDriver::Response Session::stop()
     return JsonValue {};
 }
 
+// 11.2 Close Window, https://w3c.github.io/webdriver/#dfn-close-window
+Web::WebDriver::Response Session::close_window()
+{
+    // 3. Close the current top-level browsing context.
+    TRY(web_content_connection().close_window());
+    m_windows.remove(m_current_window_handle);
+
+    // 4. If there are no more open top-level browsing contexts, then close the session.
+    if (m_windows.is_empty())
+        stop();
+
+    // 5. Return the result of running the remote end steps for the Get Window Handles command.
+    return get_window_handles();
+}
+
+// 11.3 Switch to Window, https://w3c.github.io/webdriver/#dfn-switch-to-window
+Web::WebDriver::Response Session::switch_to_window(StringView const& handle)
```

Comment:
It's better to pass StringView by value (i.e. drop the const-ref) as it fits in a register. We've gone through the code base a bunch in the past to pass these by value, e.g. 8b1108e4858f797c9216dc8ae4a3918ad50c73b4.